test/jest/playwrightEnvironment.js

```diff
@@ -114,15 +114,34 @@ class PlaywrightEnvironment extends NodeEnvironment {
     if (event.name === 'test_start') {
       const fn = event.test.fn;
       event.test.fn = async () => {
-        try {
-          return await this.fixturePool.resolveParametersAndRun(fn);
-        } finally {
-          await this.fixturePool.teardownScope('test');
-        }
+        return await this.fixturePool.resolveParametersAndRun(fn);
       };
     }
 
+    if (event.name === 'test_fn_success')
+      await this.fixturePool.teardownScope('test');
+
     if (event.name === 'test_fn_failure') {
+      let index = 0;
```

Comment:
What about timeouts?

Reply:
Timeouts are failures are they not?

Reply:
They are and I don't think your solution covers them.

Reply:
I tested it with timeouts and it correctly produces images.